docs/next/guides/building-and-testing/plugins.md

```diff
@@ -17,228 +15,242 @@ tags:
+  /**
+   * Define a unique plugin's key.
+   * Handsontable uses that string to register the plugin under that alias.
+   * Moreover, you might also use the same parameter to recognize the plugin's
+   * options passed through the Setting object at Handsontable initialization.
+   *
+   * @returns {string}
+   */
+  static get PLUGIN_KEY() {
+    return 'customPlugin';
+  }
+
+  /**
+   * Extend the default constructor and define internal properties for your plugin.
+   *
+   * @param {Handsontable} hotInstance
+   */
   constructor(hotInstance) {
+    /**
+     * BasePlugin's constructor adds to our custom plugin the following properties:
+     * - `this.hot` - A reference to the Handsontable instance, you can't overwrite this property.
+     *                It also gives you the access to columns and rows index mappers.
+     */
     super(hotInstance);
 
     // Initialize all your public properties in the class' constructor.
-    /**
-     * yourProperty description.
-     *
-     * @type {String}
-     */
-    this.yourProperty = '';
-    /**
-     * anotherProperty description.
-     * @type {Array}
-     */
-    this.anotherProperty = [];
+    this.configuration = {
+      enabled: false,
+      msg: ''
+    };
+  }
+
+  /**
+   * Unifies configuration passed to settings object.
+   *
+   * @returns {object}
+   * @throws {Error}
+   */
+  getUnifiedConfig() {
+    const pluginSettings = this.hot.getSettings()[CustomPlugin.PLUGIN_KEY];
+
+    if (pluginSettings === true) {
+      return {
+        enabled: true,
+        msg: 'default msg boolean'
+      };
+    }
+    if (Object.prototype.toString.call(pluginSettings) === '[object Object]') {
+      return {
+        enabled: true,
+        msg: 'default msg obj',
+        ...pluginSettings
+      };
+    }
+    if (pluginSettings === false) {
+      return {
+        enabled: false,
+        msg: ''
+      };
+    }
+
+    throw new Error(
+      `${CustomPlugin.PLUGIN_KEY} - incorrect plugins configuration.
+      Passed:
+        - type: ${typeof pluginSettings}
+        - value: ${JSON.stringify(pluginSettings, null, ' ')}
+
+      Expected:
+        - boolean
+        - object
+      `
+    );
   }
 
   /**
    * Checks if the plugin is enabled in the settings.
    */
   isEnabled() {
-    return !!this.hot.getSettings().internalPluginSkeleton;
+    const pluginSettings = this.getUnifiedConfig();
+
+    return pluginSettings.enabled;
   }
 
   /**
-   * The enablePlugin method is triggered on the beforeInit hook. It should contain your initial plugin setup, along with
-   * the hook connections.
-   * Note, that this method is run only if the statement in the isEnabled method is true.
+   * The `enablePlugin` method is triggered on the `beforeInit` hook.
+   * It should contain your initial plugin setup, along with the hooks connections.
+   * Note, that this method is run only if the statement in the `isEnabled` method returns true.
    */
```

Comment:
It should contain your plugin's initial setup and hook connections.
This method is run only if the \`isEnabled\` method returns \`true\`.